benchmarks/_util/websocket-simple-server.js

```diff
@@ -0,0 +1,337 @@
+'use strict'
+
+const http = require('node:http')
+const crypto = require('node:crypto')
+const stream = require('node:stream')
+
+const uid = '258EAFA5-E914-47DA-95CA-C5AB0DC85B11'
+
+class ws {
+  /**
+   * @param {number} opcode
+   * @param {Uint8Array} data
+   */
+  static createFrame (opcode, data) {
+    const length = data.length
+
+    let payloadLength = length
+    let offset = 2
+
+    if (length > 65535) {
+      offset += 8
+      payloadLength = 127
+    } else if (length > 125) {
+      offset += 2
+      payloadLength = 126
+    }
+
+    const frame = Buffer.allocUnsafeSlow(length + offset)
+
+    frame[0] = 0x80 | opcode
+
+    frame[1] = payloadLength
+
+    if (payloadLength === 126) {
+      frame.writeUInt16BE(length, 2)
+    } else if (payloadLength === 127) {
+      if (length > 0xffffffff) {
+        throw new Error('Array buffer allocation failed')
+      }
+      frame[2] = frame[3] = 0
+      frame.writeUIntBE(length, 4, 6)
+    }
+
+    if (length !== 0) {
+      frame.set(data, offset)
+    }
+
+    return frame
+  }
+
+  /**
+   * @param {Uint8Array | null} buffer
+   */
+  static parseFrame (buffer) {
```

Comment:
If you have to resort to parsing frames, it does not make much sense to use a custom server as it will not be much faster than `ws`. If you need a full featured WebSocket server I suggest to use https://github.com/uNetworking/uWebSockets.js  . A `uWS` WebSocket is faster than a plain `net.Socket`.

Reply:
Used it ;)

Reply:
thanks, custom websocket server had been dropped.